common/src/main/java/com/microsoft/hydralab/performance/parsers/AndroidBatteryInfoResultParser.java

```diff
@@ -2,17 +2,130 @@
+            List<String> contents = new ArrayList<>();
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                inputLine = inputLine.trim().toLowerCase();
+                contents.add(inputLine);
+            }
+            in.close();
+
+            //Find Uid
+            for (String line : contents) {
+                if (Pattern.matches("u0.*:", line)) {
+                    uid = line.split(":")[0];
+                }
+            }
+            if (uid == null) {
+                classLogger.error(String.format("Could not parse the battery usage of %s in battery info file: %s", packageName, rawFile.getAbsolutePath()));
+                return null;
+            }
+
+
+            for (String line : contents) {
+                //Parse total usage
+                if (mode == 1) {
+                    if (Pattern.matches("capacity:.*computed drain:.*", line)) {
+                        // find total usage
+                        totalUsage = Float.parseFloat(line.split("computed drain: ")[1].split(",")[0]);
+                        if (totalUsage == 0.0) {
+                            mode = 2;
+                        }
+                    }
+                } else {
+                    if (Pattern.matches(".*global.*", line)) {
+                        isCalculateTotalUsage = true;
+                        continue;
+                    }
+                    if (isCalculateTotalUsage && line.startsWith("uid")) {
+                        isCalculateTotalUsage = false;
+                    }
+                    if (isCalculateTotalUsage) {
+                        if (!line.startsWith("screen:")) {
+                            totalUsage += Float.parseFloat(line.split(" ")[1].trim());
+                        }
+                    }
+                }
+
+                // Parse app usage
+                if (Pattern.matches("uid " + uid + ":.*", line)) {
+                    appUsage = Float.parseFloat(line.split(": ")[1].split(" ")[0]);
+                    batteryInfo.setAppUsage(appUsage);
+                    batteryInfo.setRatio(appUsage / totalUsage);
+                    batteryInfo.setCpu(parseAppDetails(line, "cpu="));
+                    batteryInfo.setSystemService(parseAppDetails(line, "system_services="));
+                    batteryInfo.setScreen(parseAppDetails(line, "screen="));
+                    batteryInfo.setWakeLock(parseAppDetails(line, "wakelock="));
+                    batteryInfo.setWifi(parseAppDetails(line, "wifi="));
+                }
+
+                batteryInfo.setTotal(totalUsage);
+            }
+        } catch (IOException e) {
+            classLogger.error("Failed to parse the battery info file: " + rawFile.getAbsolutePath());
+            return null;
+        }
+
+        return batteryInfo;
+
+    }
+
+    float parseAppDetails(String line, String keyword) {
```

Comment:
Don't use default visibility unless you're on purpose.

Reply:
Revised